src/WebJobs.Script/Workers/Http/HttpWorkerFunctionProvider.cs

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Azure.WebJobs.Script.Workers.Rpc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers.Http
+{
+    internal class HttpWorkerFunctionProvider : IFunctionProvider
+    {
+        private const char SpaceChar = ' ';
+        private const string DoubleSlash = "//";
+
+        private readonly HttpWorkerOptions _httpWorkerOptions;
+        private readonly IHostFunctionMetadataProvider _hostFunctionMetadataProvider;
+        private readonly IOptionsMonitor<LanguageWorkerOptions> _languageWorkerOptions;
+        private readonly ILogger _logger;
+        private readonly Dictionary<string, List<string>> _errors = new();
+        private static readonly ImmutableArray<string> AllHttpMethods = ["get", "post", "put", "delete", "patch", "head", "options"];
```

Comment:
This is not an exhaustive list (e.g., missing TRACE, CONNECT, etc.), Consider using `Microsoft.AspNetCore.Http.HttpMethods` and performing validation against the methods defined there.